integration-tests/tests/client/process_blocks.rs

```diff
@@ -1558,6 +1575,44 @@ fn test_gc_after_state_sync() {
     assert!(env.clients[1].chain.clear_data(tries, 2).is_ok());
 }
 
+/// This tests the `num_epochs_to_keep_store_data` setting, which specifies the number of epochs
+/// to keep.
+///
+/// For this test, we set epoch length to `128`, and `num_epochs_to_keep_store_data` to 10.
+/// This tests shows, we keep 9 full epochs, and one partial epoch with `epoch_length`-`100` blocks.
```

Comment:
where does the 'magic' 100 come from?

if we have num_epochs_to_keep set to 10, I'd have expected all 10 epochs to be stored (and not 9.5) - unless I misunderstand this comment

Reply:
@mm-near That's a good question. I tried adjusting the value of epoch_length, let's say to 1024, and we still keep 9 full epoch + one `(epoch_length - 100)`.